### PR TITLE
[ci] wheel: let RAY_INSTALL_JAVA control java build

### PR DESCRIPTION
**Pull request by aslonnie**
previously `-n "${RAY_INSTALL_JAVA:-}"` will still compile java even when `RAY_INSTALL_JAVA` is not set to `1` (like `0`).

### CODE REVIEW

**Comment by aslonnie** [comment]
full branch ci run: https://buildkite.com/ray-project/oss-ci-build-branch/builds/5004

File: python/build-wheel-manylinux2014.sh [hunk 1]
```diff
@@ -129,7 +137,9 @@ done
 # Clean the build output so later operations is on a clean directory.
 git clean -f -f -x -d -e .whl -e python/ray/dashboard/client
 
-echo "--- Build JAR"
-if [ "${BUILD_JAR-}" == "1" ]; then
-  ./java/build-jar-multiplatform.sh linux
+if [[ "${RAY_INSTALL_JAVA}" == "1" ]]; then
+  if [[ "${BUILD_JAR-}" == "1" ]]; then
```
Comment (can-anyscale) [inline python/build-wheel-manylinux2014.sh:141]: Combine into one if?

**Review by can-anyscale — APPROVED** [review]

File: python/build-wheel-manylinux2014.sh (hunk 1 above)
Comment (aslonnie) [reply python/build-wheel-manylinux2014.sh:141]: `BUILD_JAR` is only used in linux wheels and jars build. seems that not every thing that builds java package it into a jar.

**Comment by aslonnie** [comment]
(ready to merge).

**Comment by can-anyscale** [comment]
merge!

**Pull request merged**